autobahn/wamp_invocation.hpp

```diff
@@ -29,11 +29,47 @@
 
 namespace autobahn {
 
+static const msgpack::object EMPTY_DETAILS(std::unordered_map<std::string, msgpack::object>(), nullptr);
+
+
 class wamp_invocation_impl
 {
 public:
     wamp_invocation_impl();
     wamp_invocation_impl(wamp_invocation_impl&&) = delete; // copy wamp_invocation instead
+    template <typename T>
+    T details(const std::string& key) const;
```

Comment:
Maybe you can move this declaration to underneath the doxygen comment, then both of them are right next to each other and the unnecessary empty line underneath the comment can also go away.